akka-actor-typed/src/main/scala/akka/actor/typed/internal/BehaviorImpl.scala

```diff
@@ -39,45 +38,36 @@ import scala.annotation.tailrec
   }
 
   private final case class Widened[T, U](behavior: Behavior[T], matcher: PartialFunction[U, T]) extends ExtensibleBehavior[U] {
-    @tailrec
-    private def canonical(b: Behavior[T], ctx: AC[T]): Behavior[U] = {
-      if (isUnhandled(b)) unhandled
-      else if ((b eq SameBehavior) || (b eq this)) same
-      else if (!Behavior.isAlive(b)) Behavior.stopped
-      else {
-        b match {
-          case d: DeferredBehavior[T] ⇒ canonical(Behavior.undefer(d, ctx), ctx)
```

Comment:
this case looks different in `wrapNonSpecial` ?

Reply:
This is the way it should be done, even if applying the deferred behavior is exactly what `undefer` does, I have updated now.